olive/evaluator/olive_evaluator.py

```diff
@@ -998,7 +1009,9 @@ def _evaluate_raw_latency(
     def _prepare_dataloader(self, dataloader: Dataset, model: SNPEModelHandler) -> FileListDataLoader:
         if isinstance(dataloader, FileListDataLoader):
             return dataloader
-        return FileListCommonDataLoader(dataloader, model.io_config)
+        # batch_size=1 guarantees that the dataloader returns one input at a time. And the input has
+        # the same batch_size in original dataloader.
+        return FileListCommonDataLoader(dataloader, model.io_config, batch_size=1)
```

Comment:
do we want to do this? I think using `None` might be better since we can finish the inference in one go? 
Or maybe provide an option to provide the file dataloader's batch size. Latency can provide 1 but accuracy could use `None`. 
You already implemented the logic to loop over the outputs anyways.

Reply:
I will insist as 1 to respect the batch size from original data loader.  As in logics, we give the batch_size as 2 for example, but the real inference happens in one go. Sounds confused.

Reply:
We are respecting the original batch size since that's what makes a single input to the model. The outer batch size just defines how many of them we call the net-run on. I don't see any reason not to run as many as we can at once so that we don't have the overhead of calling net-run which has to load and call the model each time. I think the net-run loops over the inputs also so it is running them one by one.

Plus, this is an internal implementation of looping over the data. I don't think we expect the user to know this anyways. It also seems to fit with your earlier comment 
![image](https://github.com/microsoft/Olive/assets/94929125/9ef6da07-6e5c-4b33-a648-31f4328769d4)

Reply:
for latency, I agree it should be 1 since we want to calculate it for one batch like we do for the other models.

Reply:
> I will insist as 1 to respect the batch size from original data loader. As in logics, we give the batch_size as 2 for example, but the real inference happens in one go. Sounds confused.

could we get the original data loader batch size? specify batch_size multiple times seems confused to us,

Reply:
Original data loader batch size cannot be assign file list data loader directly.
See this: https://github.com/microsoft/Olive/pull/925#discussion_r1477905900

Basically, file list data loader is trying to pack every iteration(batch_size, xxxx) in original data loader into a local binary file. Then user a text file(snpe_batch_size, xxx) to list the binary file as input.

Let us say here are 100 data, if original batch size is 2, then file list data loader will generate 50 binary files. 
Then if the file list batch size is 1, there will be 50 text file which contains 1 binary file path.
If the file list batch size is 10, there will be 5 text file which contains 10 binary file paths.

Reply:
I see. In this case, I think file_chunk_size is more appropriate than file_batch_size?

Reply:
yes, I think we can definitely rename this parameter both in the evaluator and the dataloader implementation to avoid confusion.

Could do it in this PR or in a follow up PR

Reply:
Upated.